bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/Variant.java

```diff
@@ -689,15 +689,20 @@ private void updateObsFlags() {
                 // custom mask, check its availability.
                 final EnumSet<Flag> maskFlags = EnumSet.noneOf(Flag.class);
                 ImOption.apply(obs.getCustomMask()).filter(m -> !m.trim().isEmpty() && shouldCheckMaskAvailability(obs)).foreach(m -> {
-                    final ProgramId pid = obs.getProg().getStructuredProgramId();
-                    if (pid instanceof ProgramId.Science) {
-                        final Availability a = owner.maskAvailability((ProgramId.Science) pid, m);
-                        if (a == Availability.SummitCabinet) {
-                            maskFlags.add(Flag.MASK_IN_CABINET);
-                        } else if (a != Availability.Installed) {
-                            maskFlags.add(Flag.MASK_UNAVAILABLE);
-                        }
+
+                    // We parse the custom mask name into a CustomMaskKey if
+                    // possible and use it to lookup the Availability.
+                    final Availability a =
+                        ImOption.fromScalaOpt(CustomMaskKey.parse(m))
+                            .map(k -> owner.maskAvailability(k))
+                            .getOrElse(Availability.Missing);
+
+                    if (a == Availability.SummitCabinet) {
+                        maskFlags.add(Flag.MASK_IN_CABINET);
+                    } else if (a != Availability.Installed) {
+                        maskFlags.add(Flag.MASK_UNAVAILABLE);
```

Comment:
No flags for `Missing`?

Reply:
Right, from the point of view of the QPT, a mask is either installed, not installed but in the summit cabinet, or else simply unavailable.  There is more detail in the ICTD app if anyone is curious.